model/diameter.py

```diff
@@ -108,6 +123,10 @@ def update_draw_offset(self, pos, ui_scale):
 
     def draw_props(self, layout):
         sub = super().draw_props(layout)
+        # self.bl_rna.properties.get('value').is_readonly = True
+        #   Traceback (most recent call last):
+        #     File "<string>", line 1, in <module>
+        #   AttributeError: bpy_struct: attribute "is_readonly" from "FloatProperty" is read-only
```

Comment:
@hlorus I have been unsuccessful at editing this attribute after the creation of the property.

Do you have knowledge on how to do this? I have not found any helpful resources for this specific case.

Reply:
I'm afraid property definitionns are not editable after registration. I would suggest to define our own getters/setters for the value properties and then just not set the value if it's a reference dimension.
https://docs.blender.org/api/current/bpy.props.html#getter-setter-example

Reply:
I was initially going to do that, but it is always preferable to sanitise user input rather than handling issues down the line ^^
Thankfully you seem to have found the way :smile: